src/main/java/io/cdap/plugin/gcp/common/GCPErrorDetailsProviderUtil.java

```diff
@@ -88,6 +110,25 @@ public static ProgramFailureException getHttpResponseExceptionDetailsFromChain(T
       String.format("%s %s: %s", errorReason, e.getClass().getName(), e.getMessage()), errorType, dependency, e);
   }
 
+  public static ProgramFailureException getApiExceptionDetailsFromChain(Throwable e, String errorReason,
+                                                                                 ErrorType errorType,
+                                                                                 boolean dependency,
+                                                                                 String externalDocUrl) {
+    List<Throwable> causalChain = Throwables.getCausalChain(e);
+    for (Throwable t : causalChain) {
+      if (t instanceof ProgramFailureException) {
+        // Avoid double wrap
+        return (ProgramFailureException) t;
+      }
+      if (t instanceof ApiException) {
+        return getProgramFailureException((ApiException) t, externalDocUrl, null);
+      }
```

Comment:
It is just another `if` condition in `getHttpResponseExceptionDetailsFromChain()`, why do we need a separate method?